diff --git a/dist/runtime/App.js b/dist/runtime/App.js
index 0672c851942fc93e07f004fde407ad5e8777d1fd..196c4e1bf5e4afa5d921b7a1fe8737ff539486eb 100644
--- a/dist/runtime/App.js
+++ b/dist/runtime/App.js
@@ -63,7 +63,7 @@ async function initPageData(routePath) {
       frontmatter: {},
       title: "404",
       toc: [],
-      version: "",
+      version: siteData.multiVersion.default || "",
       _filepath: "",
       _relativePath: ""
     }
