diff --git a/index.js b/index.js
index 9f44faaa92a1ecaa570ce686be991d322a09fe18..f9dce7905357ee113f57b5dbbb5a5137f8d428ec 100644
--- a/index.js
+++ b/index.js
@@ -160,7 +160,6 @@ var removeFlowVisitor = {
   TypeAlias: removeNode,
   TypeAnnotation: removeNodeIfNotCommentType,
   TypePredicate: removeNodeIfNotCommentType,
-  TypeParameterDeclaration: removeNode,
   InferredPredicate: removeInferredPredicateNode,
   OpaqueType: removeNode,
   DeclareOpaqueType: removeNode,
@@ -366,6 +365,12 @@ var removeFlowVisitor = {
     }
   },
 
+  TypeParameterDeclaration: function (context, node, ast, parent) {
+    if (parent.type !== 'ComponentTypeAnnotation') {
+      return removeNode(context, node);
+    }
+  },
+
   TypeParameterInstantiation: function (context, node, ast, parent) {
     // prevent interference with asExpression removal
     if (parent.type !== 'GenericTypeAnnotation') {
