diff --git a/android/CMakeLists.txt b/android/CMakeLists.txt
index 1745fb576b33dafdbe627ae4a8044353b7ddbb81..678127b97f95b933c507f4b7d74fa571a5a5156d 100644
--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -38,7 +38,16 @@ target_compile_definitions(
 
 find_package(ReactAndroid REQUIRED CONFIG)
 
-if(${RNS_NEW_ARCH_ENABLED})
+if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
+    find_package(fbjni REQUIRED CONFIG)
+ 
+    target_link_libraries(rnscreens
+      ReactAndroid::reactnative
+      ReactAndroid::jsi
+      fbjni::fbjni
+      android
+    )
+elseif(${RNS_NEW_ARCH_ENABLED})
     find_package(fbjni REQUIRED CONFIG)
     
     target_link_libraries(
diff --git a/android/src/main/jni/CMakeLists.txt b/android/src/main/jni/CMakeLists.txt
index eaab6697af04afc87a13807f97c1529d6e874b26..000c21ccd280afc75bb291cbe86335bc8f3bb8d5 100644
--- a/android/src/main/jni/CMakeLists.txt
+++ b/android/src/main/jni/CMakeLists.txt
@@ -38,25 +38,34 @@ target_include_directories(
   ${LIB_ANDROID_GENERATED_COMPONENTS_DIR}
 )
 
-target_link_libraries(
-  ${LIB_TARGET_NAME}
-  fbjni
-  folly_runtime
-  glog
-  jsi
-  react_codegen_rncore
-  react_debug
-  react_nativemodule_core
-  react_render_core
-  react_render_debug
-  react_render_graphics
-  react_render_mapbuffer
-  react_render_componentregistry
-  react_utils
-  rrc_view
-  turbomodulejsijni
-  yoga
-)
+if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 76)
+  target_link_libraries(
+    ${LIB_TARGET_NAME}
+    ReactAndroid::reactnative
+    ReactAndroid::jsi
+    fbjni::fbjni
+  )
+else()
+  target_link_libraries(
+    ${LIB_TARGET_NAME}
+    fbjni
+    folly_runtime
+    glog
+    jsi
+    react_codegen_rncore
+    react_debug
+    react_nativemodule_core
+    react_render_core
+    react_render_debug
+    react_render_graphics
+    react_render_mapbuffer
+    react_render_componentregistry
+    react_utils
+    rrc_view
+    turbomodulejsijni
+    yoga
+  )
+endif()
 
 target_compile_options(
   ${LIB_TARGET_NAME}
