diff --git a/scripts/configure-projects.js b/scripts/configure-projects.js
index 4d4bbb91bb9808f485d9afe584ed1dea78f1ed65..86a86d2e00e1f1ec3e42a12f4c3ee8b19cfb6ad0 100644
--- a/scripts/configure-projects.js
+++ b/scripts/configure-projects.js
@@ -100,14 +100,9 @@ const getRNPackageVersion = (() => {
 })();
 
 /**
- * @param {string | undefined} manifestPath
  * @returns {string | undefined}
  */
-function getAndroidPackageName(manifestPath, fs = nodefs) {
-  if (!manifestPath) {
-    return undefined;
-  }
-
+function getAndroidPackageName(fs = nodefs) {
   try {
     const rncliAndroidVersion = getRNPackageVersion(
       "@react-native-community/cli-platform-android",
@@ -128,9 +123,7 @@ function getAndroidPackageName(manifestPath, fs = nodefs) {
     // We're on 0.76 or later
   }
 
-  /** @type {{ android?: { package?: string }}} */
-  const manifest = readJSONFile(manifestPath, fs);
-  return manifest.android?.package;
+  return "com.microsoft.reacttestapp";
 }
 
 /**
@@ -179,7 +172,7 @@ function configureProjects({ android, ios, windows }, fs = nodefs) {
     config.android = {
       sourceDir,
       manifestPath,
-      packageName: packageName || getAndroidPackageName(appManifestPath, fs),
+      packageName: packageName || getAndroidPackageName(fs),
     };
 
     configureGradleWrapper(sourceDir, fs);
