diff --git a/node_modules/typedoc-plugin-markdown/dist/resources/helpers/returns.js b/node_modules/typedoc-plugin-markdown/dist/resources/helpers/returns.js
index eec500e..88f92d0 100644
--- a/node_modules/typedoc-plugin-markdown/dist/resources/helpers/returns.js
+++ b/node_modules/typedoc-plugin-markdown/dist/resources/helpers/returns.js
@@ -5,12 +5,13 @@ const models_1 = require("typedoc/dist/lib/models");
 const property_table_1 = require("./property-table");
 const type_1 = require("./type");
 function returns() {
-    const md = [`**Returns:** ${type_1.type.call(this, 'object')}`];
+    const md = [`${type_1.type.call(this, 'object')}`];
     if (this instanceof models_1.ReflectionType &&
         this.declaration &&
         this.declaration.children) {
         md.push(property_table_1.propertyTable.call(this.declaration.children));
+        md.push('\n');
     }
-    return md.join('\n\n');
+    return md.join('');
 }
 exports.returns = returns;
diff --git a/node_modules/typedoc-plugin-markdown/dist/resources/partials/main.hbs b/node_modules/typedoc-plugin-markdown/dist/resources/partials/main.hbs
index 7907d4b..f646be8 100644
--- a/node_modules/typedoc-plugin-markdown/dist/resources/partials/main.hbs
+++ b/node_modules/typedoc-plugin-markdown/dist/resources/partials/main.hbs
@@ -1,3 +1,9 @@
+<div class="toc">
+  <div class="toc-content">
+
 {{ toc }}
 
+  </div>
+</div>
+
 {{> members}}
\ No newline at end of file
diff --git a/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.signature.hbs b/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.signature.hbs
index edcbe25..81c2f3c 100644
--- a/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.signature.hbs
+++ b/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.signature.hbs
@@ -26,15 +26,13 @@
 
 {{/if}}
 
+#### Returns:
 {{#if type}}
 
 {{#with type}}{{{returns}}}{{/with}}
-
 {{#if comment.returns}}
-
 {{#with comment.returns}}
-
-{{{text}}}
+— {{{text}}}
 
 {{/with}}
 
diff --git a/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.sources.hbs b/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.sources.hbs
index d1f3c83..98ce972 100755
--- a/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.sources.hbs
+++ b/node_modules/typedoc-plugin-markdown/dist/resources/partials/member.sources.hbs
@@ -1,3 +1,4 @@
+<div class="sources">
 {{#if implementationOf}}
 
 Implementation of: {{#with implementationOf}}{{typeAndParent}}{{/with}}
@@ -33,3 +34,4 @@ Defined in: {{fileName}}:{{line}}
 {{/each}}
 
 {{/if}}
+</div>
\ No newline at end of file
diff --git a/node_modules/typedoc-plugin-markdown/dist/theme.js b/node_modules/typedoc-plugin-markdown/dist/theme.js
index 5fc0d65..bad5966 100644
--- a/node_modules/typedoc-plugin-markdown/dist/theme.js
+++ b/node_modules/typedoc-plugin-markdown/dist/theme.js
@@ -135,7 +135,7 @@ class MarkdownTheme extends theme_1.Theme {
      * @param reflection
      */
     toUrl(mapping, reflection) {
-        return mapping.directory + '/' + this.getUrl(reflection) + '.md';
+        return mapping.directory + '/' + this.getUrl(reflection);
     }
     /**
      * @see DefaultTheme.getUrl
